monkey/infection_monkey/send_all_events_to_island.py

```diff
@@ -0,0 +1,112 @@
+import logging
+import queue
+import threading
+from typing import Set
+
+import requests
+
+# TODO: shouldn't leak implementation information; can we do this some other way?
+from pubsub import pub
+
+from common.common_consts.timeouts import MEDIUM_REQUEST_TIMEOUT
+from common.events import AbstractAgentEvent
+from common.utils import Timer
+
+logger = logging.getLogger(__name__)
+
+
+DEFAULT_TIME_PERIOD = 5
+WAKES_PER_PERIOD = 4
+
+
+class send_all_events_to_island:
+    """
+    Sends information about the events carried out by the Agent to the Island in batches
+    """
+
+    def __init__(self, server_address: str):
+        self._server_address = server_address
+        self._queue: queue.Queue[AbstractAgentEvent] = queue.Queue()
```

Comment:
Can Queue be typehinted with generic in python3.7? I feel like I've tried this before and it doesn't work.

Reply:
What's wrong with doing it this way? It's how we do it in other places too.

Reply:
Nothing if it works. I'm just saying I've tried it before and had issues. Maybe `typing.Queue` isn't generic and that's what I tried to use.